src/streamsync/core.py

```diff
@@ -243,7 +243,7 @@ def apply(self, key) -> None:
 
     @staticmethod
     def escape_key(key):
-        return key.replace(".", "\.")
+        return key.replace(".", "\\.")
```

Comment:
In that case, I would advocate to use raw string instead of double quote. It remove all risk of interpret the string.

```python
return key.replace(".", r"\.")
```

https://www.geeksforgeeks.org/python-raw-strings/

Reply:
I would also for this sort of issue encourage to code a unit test that reproduce this issue, then make the fix. I should not require huge effort and it will secure advance behavior from the code.

Reply:
That's a very good idea — I was mostly using raw strings for regular expressions, and it didn't even come to mind that it could be applicable here. Thank you for noticing and suggesting it. 

Regarding unit tests, I haven't had experience with writing them specifically to catch warnings (as in this case, the behavior of program is expected => "correct"). If this was a case you've had in your practice, I'd gladly pick up an example to learn upon. 

Again, thank you for your input — it surely shows a skill when there's a room found for solid improvement even in such simple things. Glad to be working together! 🙂

Reply:
What's the use of a test here? I'm not following

Reply:
My bad for the test, I think that was a bad advice. It was to remove a warning not to fix a bug.